### PR TITLE
Handle producer errors

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/elsa/producer.ex [hunk 1]
```diff
@@ -27,63 +29,90 @@ defmodule Elsa.Producer do
...
+    reason_string =
+      "#{length(messages_sent)} messages succeeded before elsa producer failed midway through due to #{inspect(reason)}"
+
+    failed_messages = Enum.drop(message_chunks, chunks_sent) |> Enum.flat_map(fn {_partition, chunk} -> chunk end)
+    {:error, reason_string, failed_messages}
+  end
 
   defp get_client(opts) do
-    Keyword.get_lazy(opts, :name, &Elsa.default_client/0)
+    Keyword.get_lazy(opts, :client, fn -> Keyword.get_lazy(opts, :name, &Elsa.default_client/0) end)
```
Comment (jeffgrunewald) [inline lib/elsa/producer.ex]: don't you want this to be `:name` now?

Comment (jeffgrunewald) [reply lib/elsa/producer.ex]: just simplify to a single call looking for `:name` in the opts and fail over to the default client if that one's not set?

Comment (bbalser) [reply lib/elsa/producer.ex]: I was trying to maintain backward compatability, bad idea?

Comment (ManApart) [inline lib/elsa/producer.ex]: It feels weird to have two nested lazy gets, but I guess it makes sense if you're trying to create an override order/ compatibility for client or name....

Comment (jeffgrunewald) [reply lib/elsa/producer.ex]: i'd be inclined to rip off the bandaid now; seems more confusing and no one is using this yet

Comment (ManApart) [reply lib/elsa/producer.ex]: I agree with Jeff

**Pull request merged**